### PR TITLE
Added knob to control srf max n gemm blocking

### PR DESCRIPTION
**Pull request by egeor**

### CODE REVIEW

File: src/libxsmm_cpuid_x86.c [hunk 1]
```diff
@@ -495,6 +495,25 @@ LIBXSMM_API int libxsmm_cpuid_x86_amx_gemm_enforce_mx1_tile_blocking(void) {
...
+LIBXSMM_API unsigned int libxsmm_cpuid_x86_srf_gemm_set_n_max_blocking(void) {
+ unsigned  int result = 3;
+#if defined(LIBXSMM_PLATFORM_X86)
+  const char *const l_env_x86_srf_gemm_set_n_max_blocking = getenv("LIBXSMM_X86_SRF_GEMM_SET_N_MAX_BLOCKING");
+  if ( 0 == l_env_x86_srf_gemm_set_n_max_blocking ) {
+    result = 3;
+  } else {
+    if ( atoi(l_env_x86_srf_gemm_set_n_max_blocking) >= 0 ) {
+      result = (unsigned int) atoi(l_env_x86_srf_gemm_set_n_max_blocking);
+      if (result >= 8) {
```
Comment (alheinecke) [inline src/libxsmm_cpuid_x86.c:507]: Nit: >8 we auto-correct, why not auto-correcting == 0, but letting this run into an error?

File: src/generator_gemm_sse_avx_avx2_avx512.c [hunk 2]
```diff
@@ -960,6 +960,10 @@ LIBXSMM_API_INTERN unsigned int libxsmm_generator_gemm_sse_avx_avx2_avx512_get_m
     libxsmm_generator_gemm_get_blocking_and_mask( i_xgemm_desc->m, 24, 8, &l_m_blocking, &l_use_masking_a_c );
   } else if ( ( i_arch == LIBXSMM_X86_AVX )      && ( LIBXSMM_DATATYPE_F64 == LIBXSMM_GEMM_GETENUM_AB_COMMON_PREC( i_xgemm_desc->datatype ) ) ) {
     libxsmm_generator_gemm_get_blocking_and_mask( i_xgemm_desc->m, 12, 4, &l_m_blocking, &l_use_masking_a_c );
+  } else if ((i_arch == LIBXSMM_X86_AVX2_SRF) && ( ( LIBXSMM_DATATYPE_F32  == LIBXSMM_GEMM_GETENUM_C_PREC( i_xgemm_desc->datatype ) ) ||
```
Comment (alheinecke) [inline src/generator_gemm_sse_avx_avx2_avx512.c:963]: Why not FP64?

**Review by alheinecke — COMMENTED** [review]
2 minor nits

File: src/libxsmm_cpuid_x86.c (hunk 1 above)
Comment (egeor) [reply src/libxsmm_cpuid_x86.c:507]: Hm, the autocorrection logic is anyway broken here (it was not supposed to go for avx2):

https://github.com/libxsmm/libxsmm/blob/feature_config_n_blocking_srf_gemm/src/generator_gemm_sse_avx_avx2_avx512.c#L289

so I guess it has to be adjusted

File: src/generator_gemm_sse_avx_avx2_avx512.c (hunk 2 above)
Comment (egeor) [reply src/generator_gemm_sse_avx_avx2_avx512.c:963]: lemme add it there too

**Review by alheinecke — APPROVED** [review]

**Pull request merged**